docs: update reduction docs (#3501)
ansys/pyfluent: doc/source/user_guide/fields/reduction.rst

Reduction
=========

You can use reduction functions on Fluent data from one or across multiple remote Fluent sessions.
PyFluent provides both **functional** and **object-oriented** approaches to applying reduction functions.
While both are supported, the **functional approach** is emphasized for its flexibility,
particularly when working with multiple solver sessions.

Introduction to Reduction Functions
-----------------------------------

Reduction functions perform operations like computing averages, integrals, and sums over specified data locations,
such as areas or volumes.

For demonstrating the following examples, let us initialize two separate solver sessions
with two separate examples case files as follows:

.. code-block:: python

    >>> import ansys.fluent.core as pyfluent
    >>> from ansys.fluent.core.solver.function import reduction
    >>> from ansys.fluent.core.examples import download_file

    >>> solver1 = pyfluent.launch_fluent(mode=pyfluent.FluentMode.SOLVER)
    >>> case_path = download_file("Static_Mixer_main.cas.h5", "pyfluent/static_mixer")

>     >>> solver1.settings.file.read(file_type="case", file_name=case_path)

Reviewer: Please use concise form of the API whereever possible:
```
solver1 = pyfluent.launch_fluent()
solver1.settings.file.read_case(file_name=...)
```

Author: Sure, done, Thank you.

>     >>> solver1.settings.solution.initialization.hybrid_initialize()

Reviewer: In general, for any fields API we should demostrate the usage with converged data. The initialized data will have uniform distribution of pressure, so it doesn't make much sense to take average/min etc.

Author: Thanks @mkundu1, updated the docs to use converged data.

  ... )
  101325.0

**Example: Minimum Across Multiple Solvers**

.. code-block:: python

  >>> reduction.minimum(
  ...     expression="AbsolutePressure",
  ...     locations=[
  ...         solver1.setup.boundary_conditions.pressure_outlet,
  ...         solver2.setup.boundary_conditions.pressure_outlet,
  ...     ],
  ... )
  101325.0

**Example: Using Boundary Abstractions**

.. code-block:: python

  >>> reduction.minimum(
  ...     expression="AbsolutePressure",
  ...     locations=[
  ...         VelocityInlets(settings_source=solver) for solver in [solver1, solver2]
  ...     ],
  ... )
  101325.0

**Example: Geometric centroid of the velocity inlet 2**

.. code-block:: python

  >>> cent = reduction.centroid(
  >>>   locations=[solver.settings.setup.boundary_conditions.velocity_inlet["inlet2"]]
  >>> )
  >>> cent.array
  array([-0.00100001, -0.003     ,  0.00150005])

**Example: Geometric centroid of the velocity inlets over multiple solvers**

.. code-block:: python

  >>> cent = reduction.centroid(
  >>>   locations=[VelocityInlets(settings_source=solver) for solver in [solver1, solver2]]
  >>> )

>   >>> cent.array

Reviewer: I'd prefer a simple Python list over a structure behind a private class.

The docstring of centroid says it returns a float.

Author: Done, Thank you.

Reduction
=========

You can use reduction functions on Fluent data from one or across multiple remote Fluent sessions.
PyFluent provides both **functional** and **object-oriented** approaches to applying reduction functions.
While both are supported, the **functional approach** is emphasized for its flexibility,
particularly when working with multiple solver sessions.

Introduction to Reduction Functions
-----------------------------------

Reduction functions perform operations like computing averages, integrals, and sums over specified data locations,
such as areas or volumes.

> For demonstrating the following examples, let us initialize two separate solver sessions

Reviewer 2: ```suggestion
To demonstrate the following examples, first initialize two separate solver sessions
```
(Vale warning)
